src/contracts/definitions/snip-721-def.ts

```diff
@@ -513,6 +516,179 @@ export const snip721Def: ContractDefinition = {
+      token_id: string,
+      sendViewer?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        private_metadata: {
+          token_id,
+        },
+      };
+
+      if (!sendViewer) return query;
+
+      return { with_permit: { query, permit } };
+    },
+
+    getNftDossier(
+      { permit }: Context,
+      token_id: string,
+      include_expired?: boolean,
+      sendViewer?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        nft_dossier: {
+          token_id,
+          include_expired,
+        },
+      };
+      if (!sendViewer) return query;
+
+      return { with_permit: { query, permit } };
+    },
+
+    getTokenApprovals(
+      { permit }: Context,
+      token_id: string,
+      include_expired?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        token_approvals: {
+          token_id,
+          include_expired,
+        },
+      };
+
+      return { with_permit: { query, permit } };
+    },
+
+    getApprovedForAll(
+      { address: owner, permit }: Context,
+      include_expired?: boolean,
+      sendViewer?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        approved_for_all: {
+          owner,
+          include_expired,
+        },
+      };
+
+      if (!sendViewer) return query;
+
+      return { with_permit: { query, permit } };
+    },
+
+    getInventoryApprovals(
+      { address, permit }: Context,
+      include_expired?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        inventory_approvals: {
+          address,
+          include_expired,
+        },
+      };
+
+      return { with_permit: { query, permit } };
+    },
+
+    getTokens(
+      { address: owner, permit }: Context,
+      viewer?: string,
+      start_after?: string,
+      limit?: number,
+      sendViewer?: boolean
+    ): ContractQueryRequest {
+      const query = {
+        tokens: {
+          owner,
+          viewer,
+          start_after,
+          limit,
+        },
+      };
+
+      if (!sendViewer) return query;
+
+      return { with_permit: { query, permit } };
+    },
+
+    getTransactionHistory(
+      { address, permit }: Context,
+      page?: number,
+      page_size?: number
+    ): ContractQueryRequest {
+      const query = {
+        transaction_history: {
+          address,
+          page,
+          page_size,
+        },
+      };
+
+      return { with_permit: { query, permit } };
+    },
+
+    getAllTokens({ permit }: Context, limit?: number, sendViewer?: boolean) {
+      const query = { all_tokens: { limit } };
+
+      if (!sendViewer) return query;
+
+      return { with_permit: { query, permit } };
+    },
+
+    getVerifyTransferApproval({ permit }: Context, token_ids: Array<string>) {
+      const query = { verify_transfer_approval: { token_ids } };
+
+      return { with_permit: { query, permit } };
+    },
+  },
+};
+
+export const snip721DefPermit = extendContract(snip721Def, snip721Permit);
```

Comment:
I would say that as a good practice to add the `Def` suffix to the name of a contract definition. So in this case should be `snip721PermitDef`.